x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizer.java

```diff
@@ -405,4 +416,15 @@ private Tuple<List<Attribute>, List<Stat>> pushableStats(AggregateExec aggregate
         }
     }
 
+    public static boolean hasIdenticalDelegate(FieldAttribute attr, SearchStats stats) {
```

Comment:
I feel like this should return the `MappedFieldType` for the identical delegate. Or something like that. Some identifier.

Reply:
It's possible, but it would require a bit of refactoring in the pushdown rules to really take advantage of the returned `MappedFieldType`, and it's not a trivial change. I think we can consider it for a follow-up

Reply:
Cool. We just have to make sure the sub-field it finds is the right one. In case there is more than one candidate. Evil, I know.

Reply:
We are safe in this sense, if there are two subfields, none is used, even if one of them is good (it's QL logic, not optimal, but in this case it makes our life easier)